exporter/exporterhelper/common.go

```diff
@@ -51,37 +100,121 @@ func WithStart(start Start) ExporterOption {
 	}
 }
 
+// WithShutdown overrides the default TimeoutSettings for an exporter.
+// The default TimeoutSettings is 5 seconds.
+func WithTimeout(timeout TimeoutSettings) ExporterOption {
+	return func(o *baseExporter) {
+		o.cfg.TimeoutSettings = timeout
+	}
+}
+
+// WithRetry overrides the default RetrySettings for an exporter.
```

Comment:
It will retry the same request until either the exporter returns a permanent error, or the max elapsed time expires. Will not add it back to the queue, because we should try to preserve the order of exporting as much as possible.